### ISSUE TITLE
Specify different batch_size for training and validation data in fit()

### ISSUE THREAD

**Issue by jing199301** [keras-team/keras#11084]
I organize every batch data in a specific way, so I hope to use fit() in every epoch training like this:

for i in range(10):    #10 epochs
    model.fit(train_data,train_label, batch_size=126, epochs=1, validation_data=(valid_data, valid_label), shuffle=False)

where the training data is set in size of 126, then I set the batch_size=126 and epochs=1, hope to calculate only one time and update the model.
But the size of validate_data is 192. When it turns to the validate phase, it reports the error:

![xnip2018-09-05_17-06-10](https://user-images.githubusercontent.com/14290267/45083355-2c05f500-b12e-11e8-9d10-580d3234d8d4.jpg)

I notice that the input shape 66=192-126, so I assume that the fit() function takes the same batch_size for training and validation data. So I change the size of validation data to 126, the fit() function works normally.
So I wonder if it is a way to specify different batch_size for training and validation data in fit()?

**Comment by bileschi** [comment]
Can you call model.fit without the validation_data and then model.evaluate afterwards?  You could then use the batch_size parameter to evaluate.

**Comment by jing199301** [comment]
Thanks for replying!
 But I want to use 'callback' function. Could it be possible if I use callback function using my own loss monitor(such as 'my_own_loss') without validation_data?

**Comment by gabrieldemarmiesse** [comment]
I believe the easiest solution is to use `fit_generator`. You'll have one generator for training and one generator for testing, so you'll be able to customize the batch size easily.
